aten/src/ATen/core/function_schema.h

```diff
@@ -199,6 +208,28 @@ struct FunctionSchema {
       const FunctionSchema& old,
       std::ostream* why_not = nullptr) const;
 
+  // Checks whether this schema is forward compatible with the old one.
```

Comment:
Cool list and super clear. Might be helpful to actually state which case is allowed/supported, too, which is adding a new kwarg with a default value at the end of a signature without out= or just before the out= kwarg on signatures that have it

Reply:
done